cpp/src/io/statistics/typed_statistics_chunk.cuh

```diff
@@ -202,11 +202,19 @@ __inline__ __device__ typed_statistics_chunk<T, include_aggregate> block_reduce(
   using E              = typename detail::extrema_type<T>::type;
   using extrema_reduce = cub::BlockReduce<E, block_size>;
   using count_reduce   = cub::BlockReduce<uint32_t, block_size>;
+#if CCCL_MAJOR_VERSION >= 3
```

Comment:
Do we have all the includes we need for cuda::maximum?

Do includes need to be conditional on CCCL version, too?

Reply:
I pulled the definition into a separate header that pulls in all the required functionality